apex/contrib/csrc/fmha/src/fmha/philox.h

```diff
@@ -0,0 +1,96 @@
+/***************************************************************************************************
+ * This code was copied from APEX (https://github.com/NVIDIA/apex). 
+ *
+ * APEX is released under BSD-3 license.
+ *
+ **************************************************************************************************/
+#pragma once
+// Philox CUDA.
+
+class Philox {
```

Comment:
do we need a copy of this? CC @mcarilli

Reply:
Looks like it's modified to do a "7-round" philox. Pytorch and curand do a "10-round" philox. No idea what the mathematical implications are, but this doesn't do the same thing as pytorch and curand philox. If they want it to do the 7 round thing they need their own copy.

Reply:
This also returns float4, whereas the original code returns float.

Reply:
I forgot to remove it. It's actually coming from here: https://github.com/NVIDIA/apex/blob/master/apex/contrib/csrc/multihead_attn/philox.h

Reply:
Right, and that one is coming from pytorch philox that @mcarilli  was referring to above.